article/results.tex

```diff
@@ -56,59 +56,75 @@ \subsection{Best Practice Guidelines}
 As part of this work we have contributed substantial changes to the original OPFVTA repository, based on which we formulate a number of best practice guidelines, highly relevant in the production of reexecutable research outputs.
 
 \subsubsection{Errors should be fatal more often than not}
-The high complexity of a full analysis workflow makes missing output data intractable, as there are many steps at which data may have been dropped.
-Manually setting individual invocations propagate exit codes throughout the Make system is cumbersome, and thus an easy global invocation should be relied on so that errors become fatal and immediately visible.
-This is best accomplished by prepending all Bash scripts for which this concern is relevant with a \texttt{set -eu} line.
+
+By default any Python script or a compiled C or C++ program just errors out and exits immediately when running into some unexpected operation.
+Shell scripts behave differently.
+By default they continue with execution of the next scripted command, and only when the list of commands is exhausted or explicit `exit` command is called, it finally exits with a non-0 error code.
+As a result, even if not causing some detrimental actions, an execution of the script could continue for hours before it fails, and original error message might be lost in the flood of output making it hard or impossible to localize original problem to be fixed.
+Analogously, shell scripts treat undefined variables as a variable having an empty value, and thus not causing any error at all.
+As a result a command in a script like \texttt{rm -rf "\${PREFIX}/"} would do its best to try to remove everything on your harddrive in case \texttt{PREFIX} variable is not defined.
+\texttt{set -u} would change the default behavior to error out and stop execution as soon as undefined variable is attempted to be used.
```

Comment:
Did we actually ever encounter this issue? I am pretty sure we only encountered the non-fatal erroring issue. Also, we mention `set -u` explicitly, which might be the more theoretical contribution (in light of us not having really encountered it), and describe `set -e` only implicitly.

Reply:
I don't know if we did... did we? if writing was not postponed and we added direct references -- we might have known.  But I know that I did encounter this issue MANY times in other scripts.  
So, is there anything to change?

Reply:
Yes, but I can change it after merge if you want. If you want to change it basically we should introduce `set -e` explicitly and as the main thing, and only mention `set -u` after that. I'm pretty sure we never had that issue with this article. You just put `set -eu` directly and I didn't initially even know what the `u` was for.

Reply:
I guess we could do that, or just point to the sample `set -eu` stating that it is just a good safer default to use.